From aa8a1ba635978f9fad305dd7c113341bceaad6d1 Mon Sep 17 00:00:00 2001
From: Gao Pan <pandy.gao@nxp.com>
Date: Mon, 17 Jul 2017 16:57:48 +0800
Subject: [PATCH 2146/5242] MLK-15995 imx: mlb: only use irq_ahb1 for imx6

commit  2c516cad3e11fcb6d4039b0375de637249166b10 from
https://source.codeaurora.org/external/imx/linux-imx.git

Due to IP integration difference, there are 2 ahb irqs
for imx6 and only 1 ahb irq for imx8. This patch makes
mlb driver compatible with irq difference.

Signed-off-by: Gao Pan <pandy.gao@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mxc/mlb/mxc_mlb.c |   18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/drivers/mxc/mlb/mxc_mlb.c b/drivers/mxc/mlb/mxc_mlb.c
index cebcf5b..2699bb0 100755
--- a/drivers/mxc/mlb/mxc_mlb.c
+++ b/drivers/mxc/mlb/mxc_mlb.c
@@ -1565,12 +1565,14 @@ static void mxc_mlb150_irq_enable(struct mlb_data *drvdata, u8 enable)
 {
 	if (enable) {
 		enable_irq(drvdata->irq_ahb0);
-		enable_irq(drvdata->irq_ahb1);
 		enable_irq(drvdata->irq_mlb);
+		if (drvdata->irq_ahb1 > 0)
+			enable_irq(drvdata->irq_ahb1);
 	} else {
 		disable_irq(drvdata->irq_ahb0);
-		disable_irq(drvdata->irq_ahb1);
 		disable_irq(drvdata->irq_mlb);
+		if (drvdata->irq_ahb1 > 0)
+			disable_irq(drvdata->irq_ahb1);
 	}
 }
 
@@ -2622,15 +2624,13 @@ static int mxc_mlb150_probe(struct platform_device *pdev)
 
 	/* ahb1 irq */
 	drvdata->irq_ahb1 = platform_get_irq(pdev,  2);
-	if (drvdata->irq_ahb1 < 0) {
-		dev_err(&pdev->dev, "No ahb1 irq line provided\n");
-		goto err_dev;
-	}
 	dev_dbg(&pdev->dev, "ahb1_irq: %d\n", drvdata->irq_ahb1);
-	if (devm_request_irq(&pdev->dev, drvdata->irq_ahb1, mlb_ahb_isr,
+	if (drvdata->irq_ahb1 > 0) {
+		if (devm_request_irq(&pdev->dev, drvdata->irq_ahb1, mlb_ahb_isr,
 				0, "mlb_ahb1", NULL)) {
-		dev_err(&pdev->dev, "can't claim irq %d\n", drvdata->irq_ahb1);
-		goto err_dev;
+			dev_err(&pdev->dev, "can't claim irq %d\n", drvdata->irq_ahb1);
+			goto err_dev;
+		}
 	}
 
 	/* mlb irq */
-- 
1.7.9.5

